bin/propolis-server/src/lib/server.rs

```diff
@@ -271,6 +271,8 @@ fn instance_spec_from_request(
         instance_spec::SerialPortNumber::Com1,
         instance_spec::SerialPortNumber::Com2,
         instance_spec::SerialPortNumber::Com3,
+        // SoftNpu uses this port for ASIC management.
+        #[cfg(not(feature = "falcon"))]
         instance_spec::SerialPortNumber::Com4,
```

Comment:
Lingering question for @gjcolombo: Is there a future where we might want to attach serial ports to things other than a user console?  In such a case, where COM port attachment is described in the instance spec, we could probably avoid its omission here (in the falcon case), and just spell out that it's being used by the SoftNPU?

Reply:
I also wonder if there's a device id we could "borrow" for a PCI-attached 16650 chip, which the SoftNPU could just attach at slot X function 1, where that exposed topology implied the connection between the UART and the NPU stuff.

Reply:
> Lingering question for @gjcolombo: Is there a future where we might want to attach serial ports to things other than a user console? In such a case, where COM port attachment is described in the instance spec, we could probably avoid its omission here (in the falcon case), and just spell out that it's being used by the SoftNPU?

I'd model that as a device/backend split: the device handles all the UART register emulation, and the bytes it receives/transmits are sent to/obtained from some backend, which could be a human-readable I/O channel of some kind, a SoftNPU device, etc. I'd have to look at the serial port code again (it's been a while) to see how big of a refactoring that would be.

This will also be sort of relevant when #231 lands, because it lets callers like Omicron and PHD pass instance specs directly to Propolis, and they won't take this path. But I don't think those consumers plan to use Falcon builds, if they do they probably won't attach a SoftNPU device, and if they do we can catch/resolve the conflict in Propolis (today I'm pretty sure it'll panic because two devices will try to map the same part of PIO space, but we can make that cleaner by checking the instance spec for a conflict). So I don't think we need to refactor anything right away or change anything in this PR (though I might want to touch up #231 a bit after this lands).

I've filed #259 to track the broader "serial ports should have devices and backends" idea.